### ISSUE TITLE
Automation parameters

### ISSUE THREAD

**Issue by grammoboy2** [Geonkick-Synthesizer/geonkick#140]
At least in Carla it seems you can't automate parameters. I think that would be a important feature to add.

**Comment by iurienistor** [comment]
Yes, Geonkick does not support automation of parameters. For now I don't know when I'll start to work on this. :)

**Comment by unfa** [comment]
I guess the difficult thing is - Geonkick would need to resynthesize the drums samples at every change of any automatable parameter, or develop a completely different synthesis engine, where it's all done in real-time and not pre-cached as a sample.

If that can be done without sacrificing the envelope evaluation quality (most important for punchy drums!) I think it'd be fantastic, but honestly I very rarely need to automate anything about my drum patches, and if I do - I can always use a different synthesizer for that sound.

From my perspective this would probably be a very labor-intensive feature to implement and not sure if' I'd be worth the hassle.

What do you think?

**Comment by grammoboy2** [comment]
> What do you think?
>
To take the power-user-number-one's opinion very seriously. ;)
>

**Comment by iurienistor** [comment]
@unfa 
1. Automation of parameters can be implemented even if in the actual situation because anyway when the user changes a parameter the result is heard only if the key is pressed again. I'll start adding automation of parameters from the next releases. This actually should be related to changes in the LV2 part of the code rather than the synth.

2. But in order to be able to change the parameters in the middle of playing key than there is a need to change things in the synth. 
But usually with percussive sounds you often want say the next beat to sound brighter than the previous one, but not half brighter and half not. Ok, you might do this for very long kicks...

Thus, for 1 option automation can work.

**Comment by iurienistor** [comment]
If anyone interested in developing this feature, please, leave a comment and I'll assign the issue.
